src/services/utilServices/SgidAuthService.ts

```diff
@@ -66,17 +81,28 @@ export default class SgidAuthService {
     )
   }
 
-  retrieveSgidUserEmail(
+  retrieveSgidUserEmails(
     accessToken: string,
     sub: string
-  ): ResultAsync<string | undefined, SgidFetchUserInfoError> {
+  ): ResultAsync<PublicOfficerData[] | undefined, SgidFetchUserInfoError> {
     return ResultAsync.fromPromise(
       this.sgidClient
         .userinfo({
           accessToken,
           sub,
         })
-        .then(({ data }) => data[SGID_WORK_EMAIL_SCOPE]),
+        .then(({ data }) => {
+          const employmentDetails: SgidPublicOfficerDetails[] = JSON.parse(
+            data[SGID_WORK_EMAIL_SCOPE]
+          )
+          const employmentResp = employmentDetails.map((detail) => ({
+            email: detail.work_email,
+            agencyName: detail.agency_name,
+            departmentName: detail.department_name,
+            employmentTitle: detail.employment_title,
+          }))
+          return employmentResp
```

Comment:
supernit: can just combine (not imporant to fix)

Reply:
Split because i needed to cast earlier!